modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/authcode/UpdatingAuthorizationCodeState.java

```diff
@@ -55,14 +60,22 @@ private void update(ResourceOwnerOAuthContext context) {
         onUpdate.accept(context);
       }
     });
+    this.dancer = dancer;
+    this.config = config;
+    this.muleContext = muleContext;
   }
 
   @Override
   public String getAccessToken() {
-    if (invalidated) {
-      throw new TokenInvalidatedException(
-                                          "OAuth token for resource owner id " + delegate.getResourceOwnerId()
-                                              + " has been invalidated");
+    if (invalidated || !muleContext.getClusterId().isEmpty()) {
+      try {
+        dancer.accessToken(getResourceOwnerId());
```

Comment:
i don't understand this change, it fundamentally changes behavior. If the token was invalidated, then it should fail as it did before. Now it only fails if the state cannot be updated. Why call `dancer.accessToken()` ? Why update the underlying state? I would expect the local state to have been updated the moment the token was invalidated

Reply:
Changed it back to if (invalided), and then added if (isClusterEnabled) for the other nodes in the same cluster that couldn't get the invalidation information (due to the same key ResourceOwnerId, Dancer will have only one listener). The only way to update the UpdatingAuthorizationCodeState in this case is by getting the information from Dancer.